### ISSUE TITLE
jack midi: Program Change events not sent

### ISSUE THREAD

**Issue by Animtim** [ahlstromcj/sequencer64#75]
I just noticed that Program Change events are not sent when using native jack midi. They work fine with alsa midi.

I also quickly checked the other special events, they are correctly sent except for Channel Pressure.

**Comment by ahlstromcj** [comment]
I pushed a fix to wip for that, was accidentally ignoring 2-byte MIDI events.  Haven't had time to test it.  Thanks for all these reports!  Nothing like clean code :-).

**Comment by Animtim** [comment]
Tested, but issue is not fixed: still no Program change and Channel Pressure sent with jack midi (I even checked with the "Midi activity" window in ardour to be sure).

**Comment by ahlstromcj** [comment]
Hmmm.  I did a quick check, connecting yoshimi and jack-keyboard to seq64.  I verified that the MIDI file playback dumps its program change to yoshimi, while I can have seq64 forward a program change from jack-keyboard to yoshimi if "thru" is turned on.  I will do a more extensive check tonight.

**Comment by ahlstromcj** [comment]
Tested at home with gmidimonitor, and I see that the program changes are being received and are thru'd to the output.  The one thing I am not seeing is that they are being recorded, although notes do get recorded.  Investigating tonight.

**Comment by Animtim** [comment]
Ok, indeed it works connected to standalone yoshimi, but not with yoshimi inside ardour, which works when sending alsa midi through a2jmidid.
I guess it is now a bug with ardour not receiving those events properly from jack midi.

**Comment by ahlstromcj** [comment]
I'm still testing.  I discovered that the automatic ALSA connection between VMPK and seq64 or seq24, with "dump midi" and recording active gives a nice feedback loop that overwhelms the system, just like in the old days looping a keyboard's output to it input :-D.  It no longer crashes seq64 (I pushed that fix to wip), but after awhile, even closing all the apps leaves a note playing until laptop shutdown.  Screws up ALSA!
